internal/sessioncache/key_orig.go

```diff
@@ -0,0 +1,39 @@
+package sessioncache
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type OrigKey struct {
+	ProfileName string
+	ProfileConf map[string]string
+	Duration    time.Duration
+}
+
+// Key returns a key for the keyring item. This is a string containing the source profile name,
+// the profile name, and a hash of the duration
+//
+// this is a copy of KeyringSessions.key and should preserve behavior, *except* that it assumes `profileName`
+// is a valid and existing profile name
+func (k OrigKey) Key() string {
+	// nick: I don't understand this at all. This key function is roughly:
+	// sourceProfileName + hex(md5(duration + json(profileConf)))
+	// - why md5?
+	// - why the JSON of the whole profile? (especially strange considering JSON map order is undetermined)
```

Comment:
Do we also want another implementation of the `Key` interface in this PR to play around with, or maybe that should come later

Reply:
In the new tests, I've got a thing called `fixedKey` that just returns the same string.

If you're thinking about actual, potentially useful ones, I think we'd need to understand what this cache is supposed to accomplish first ;)